### ISSUE TITLE
env gem home incorrect on ubuntu 16.04 running in VMWare

### ISSUE THREAD

**Issue by pitosalas** [rbenv/rbenv#1021]
I've looked at many issues describing this, but no solution yet. In installed ubuntu 16.04 over VMware over Mac..

rbenv init says to install these lines in .bash_profile. But on ubuntu .bash_profile isn't run. So I put those lines at the end of .bashrc:

export PATH="$HOME/.rbenv/bin:$PATH"
eval "$(rbenv init -)"

Everything runs nicely, ruby is installed, gem is installed in the right place, and yet gem env home is incorrect.

```
pitosalas@ubuntu:~$ which ruby
/home/pitosalas/.rbenv/shims/ruby
pitosalas@ubuntu:~$ which gem
/home/pitosalas/.rbenv/shims/gem
pitosalas@ubuntu:~$ gem env home
/var/lib/gems/2.3.0
pitosalas@ubuntu:~$ 

```
In other words, gem's home is not set right, and so I cannot install gems without sudo (which I don't want to do.)

I know I can do this: `export GEM_HOME=/home/pitosalas/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/`

But I want to give working instructions to my students, so it would be great to see what I did wrong to get to this or whether this is a known issue.

**Comment by tsujigiri** [comment]
I have the same issue.

**Comment by jrcharney** [comment]
Same problem on Bash for Ubuntu on Windows in wsltty with Zsh. Not even rbenv-build or rbenv-gemset (#879) can resolve. Gem keeps pointing to `/usr/bin/gem` when calling `rbenv which gem` and `/var/lib/gems/2.3.0` when calling `gem env home`.

```
$ rbenv which gem
/usr/bin/gem
$ gem env home
/var/lib/gems/2.3.0
$ whereis ruby
ruby: /usr/bin/ruby /usr/bin/ruby2.3 /usr/lib/ruby /home/Jason/.rbenv/shims/ruby /usr/share/man/man1/ruby.1.gz
$ which ruby
/home/Jason/.rbenv/shims/ruby
$ whereis gem
gem: /usr/bin/gem /usr/bin/gem2.3 /home/Jason/.rbenv/shims/gem /usr/share/man/man1/gem.1.gz
$ which gem
/home/Jason/.rbenv/shims/gem
```

Hmm...**AH HA! figured it out!**  We didn't do a `rbenv rehash` after installing things!

(BTW, if you haven't run `rbenv global 2.5.0` (or whatever version you plan on using in the future) after doing `rbenv install -v 2.5.0`, please do so. That might help too.)

Check this out, this happened after I tried those commands above

```
$rbenv rehash
$ rbenv which gem
/home/Jason/.rbenv/versions/2.5.0/bin/gem
$ gem env home
/home/Jason/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0
$ whereis ruby
ruby: /usr/bin/ruby /usr/bin/ruby2.3 /usr/lib/ruby /home/Jason/.rbenv/shims/ruby /usr/share/man/man1/ruby.1.gz
$ which ruby
/home/Jason/.rbenv/shims/ruby
$ whereis gem
gem: /usr/bin/gem /usr/bin/gem2.3 /home/Jason/.rbenv/shims/gem /usr/share/man/man1/gem.1.gz
$ which gem
/home/Jason/.rbenv/shims/gem
```

The output for the `whereis` and `which` commands didn't change, but the command  output for `rbenv which gem` and `gem env home` did! And that's what should happen.  Because once that is done, we can install Bundler. **(Remember to rehash!)**

```
$gem install bundler
$rbenv hash
```

Shout outs to http://www.enhgo.com/code/setup-ubuntu-16.04-for-android-development/ and https://gorails.com/setup/ for clearing things up.

**Comment by tsujigiri** [comment]
I was able to solve my case with `rm -rf ~/.bundle`. :man_shrugging:

**Comment by mislav** [comment]
Doing `rbenv global 2.4.1` (or `rbenv local 2.4.1` inside your project) and then `rbenv rehash` should result in the correct gem home being printed when running `gem env` after that.
